osfoffline/filesystem_manager/sync_local_filesystem_and_db.py

```diff
@@ -11,209 +13,185 @@
+def get_sha256(path, chunk_size=1024**2):
+    s = hashlib.sha256()
+    with open(path, 'rb') as fobj:
+        while True:
+            chunk = fobj.read(chunk_size)
+            if not chunk:
+                break
+            s.update(chunk)
+    return s.hexdigest()
 
 
 class LocalDBSync(object):
     COMPONENTS_FOLDER_NAME = 'Components'
 
-    def __init__(self, absolute_osf_dir_path, observer, user):
-        if not isinstance(observer, Observer):
-            raise TypeError
-        if not isinstance(user, User):
-            raise TypeError
-        if not os.path.isdir(absolute_osf_dir_path):
+    def __init__(self, observer, user, loop=None):
+        self.user = user
+        self.observer = observer
+        self.loop = loop or asyncio.get_event_loop()
+        self.osf_query = OSFQuery(self.loop, self.user.oauth_token)
+        self.osf_folder = self.user.osf_local_folder_path
+
+        if not os.path.isdir(self.osf_folder):
             raise FolderNotInFileSystem
 
-        self.osf_path = ProperPath(absolute_osf_dir_path, True)
-        self.observer = observer
-        self.user = user
+        self.osf_path = ProperPath(self.osf_folder, True)
 
-    def emit_new_events(self):
-        for local_child, db_child in self._make_local_db_tuple_list(self.osf_path, self.user):
-            self._emit_single_event_and_recurse(local_child, db_child)
+    def sync_db(self):
+        logger.info('Beginning initial sync')
+        events, nodes = [], [
+            node for node in
+            self.user.top_level_nodes
+            if node.osf_id in self.user.guid_for_top_level_nodes_to_sync
+        ]
+        for node in nodes:
+            logger.info('Resyncing node {}'.format(node))
+            events.extend(self.loop.run_until_complete(self.crawl_node(node)))
+        logger.info('Initial sync finished')
+
+    def match_local_remote(self, local_list, remote_list):
+        ret = {}
+        for local in local_list:
+            ret.setdefault((local.name, local.is_dir), [None, None])[0] = local
+
+        for remote in remote_list:
+            ret.setdefault((remote.name, remote.is_dir), [None, None])[1] = remote
+
+        return ret
+
+    @asyncio.coroutine
+    def crawl_node(self, node, local_path='/', remote_path='/'):
+        logger.debug('Crawling node {} at path {}'.format(node, local_path))
+        events, directories = [], []
+        resources = self.match_local_remote(
+            self.list_local_dir(node, local_path),
+            (yield from self.list_remote_dir(node, remote_path))
+        )
+
+        for (name, is_dir), (local, remote) in resources.items():
+            if is_dir:
+                directories.append((
+                    getattr(local, 'fullpath', None),
+                    getattr(remote, 'osf_id', None)
+                ))
+            event = self.infer_event(local, remote)
+            if event is not None:
+                events.append(event)
+                # DO SOMETHING HERE
```

Comment:
This class doesn't seem to have a queue at all; are the comments on these lines still relevant in any way?